module4/lessons/Week3/MaintainingState.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: page
+title: Maintaining State
+---
+
+## Learning Goals
+* Understand how cookies impact our experience on the internet
+* Implement a simple cookie
+* Recognize some limitations and regulations with cookies
+
+## Warm Up
+
+<section class='call-to-action' markdown='1'>
+
+**With a Partner**  
+Based on your experience interacting with websites:
+* What is a cookie?
+* How do you know if a website uses cookies?
+
+</section>
+
+
+## If you Give a Dev a Cookie
+
+HTTP is a stateless protocol.  This means that each request to a server exists in isolation - the server makes no connection between one request and the next. This makes for a very boring internet!  If we stayed in a stateless environment, we wouldn't be able to log in to applications, or save personal preferences, and the online shopping experience would be much more difficult.
+
+<section class='call-to-action' markdown='1'>
+
+**Independent Research**
+Spend 20 minutes researching cookies in web development.  Try to answer the following questions:
+* In technical terms, what is a cookie?
```

Comment:
Might be good to have a question here about why a website might want to use a cookie. I think seems clear to us, but maybe not to students.

Reply:
Or I see you have it in the intro section above. That might be enough. Just want to make sure everyone is on the same page about why we have a cookies and what they do for us before digging into the what and how.